src/factories/node.ts

```diff
@@ -14,12 +14,15 @@ export type NodeContainerFactory = Awaited<ReturnType<typeof nodeContainerFactor
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
 export async function nodeContainerFactory(node: RunGraphNode) {
+  const application = await waitForApplication()
   const cull = await waitForCull()
   const { animate } = await animationFactory()
   const { element: container, render: renderNode, bar } = await getNodeFactory(node)
-  const cacheKey: string | null = null
 
+  let internalNode = node
```

Comment:
Keeping an internal value of the node that gets updated every time render is called. This is used to re-render internally.